File: frontend/src/types/appointment.ts
export type AppointmentStatus = "SCHEDULED" | "COMPLETED" | "CANCELLED";

export interface Appointment {
  id: string;
  patient_id: string;
  scheduled_by: string;
  title: string;
  description: string | null;
  appointment_date: string;
  status: AppointmentStatus;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface AppointmentCreate {
  patient_id: string;
  title: string;
  description?: string | null;
  appointment_date: string;
  notes?: string | null;
}

export interface AppointmentUpdate {
  title?: string;
  description?: string | null;
  appointment_date?: string;
  notes?: string | null;
  patient_id?: string;
}

export interface AppointmentStatusUpdate {
  status: AppointmentStatus;
}

export interface AppointmentListResponse {
  items: Appointment[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export interface AppointmentFilters {
  page?: number;
  size?: number;
  status?: AppointmentStatus;
  date_from?: string;
  date_to?: string;
  patient_id?: string;
}

File: backend/app/schemas/appointment.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from typing import Optional

from app.models.appointment import AppointmentStatus


class AppointmentCreate(BaseModel):
    patient_id: uuid.UUID
    title: str
    description: Optional[str] = None
    appointment_date: datetime
    notes: Optional[str] = None


class AppointmentUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    appointment_date: Optional[datetime] = None
    notes: Optional[str] = None
    patient_id: Optional[uuid.UUID] = None


class AppointmentStatusUpdate(BaseModel):
    status: AppointmentStatus


class AppointmentResponse(BaseModel):
    id: uuid.UUID
    patient_id: uuid.UUID
    scheduled_by: uuid.UUID
    title: str
    description: Optional[str] = None
    appointment_date: datetime
    status: AppointmentStatus
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AppointmentListResponse(BaseModel):
    items: list[AppointmentResponse]
    total: int
    page: int
    size: int
    pages: int
